modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java

```diff
@@ -352,13 +355,27 @@ protected void beforeCreateDocument(DocumentModel doc) {
         snapDoc = session.saveDocument(snapDoc);
         snapDoc = session.move(snapDoc.getRef(), session.getParentDocumentRef(tmp.getRef()), null);
 
+        commitOrRollbackAndRestartTransaction();
+        try {
+            Framework.getService(BulkService.class).await(Duration.ofMinutes(10));
```

Comment:
Hoping there is not a many days until completion command running on the same instance.